sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,196 @@
+    assert len(ownership_list) == len(newownershiplist)
+    for i in range(len(newownershiplist)):
+        assert ownership_list[i]['etag'] == newownershiplist[i]['etag']
+        assert ownership_list[i]['last_modified_time'] == newownershiplist[i]['last_modified_time']
+
+def _update_and_list_checkpoint(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    partition_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == 0
+    for i in range(partition_cnt):
+            checkpoint = {
+                'fully_qualified_namespace': fully_qualified_namespace,
+                'eventhub_name': eventhub_name,
+                'consumer_group': consumer_group,
+                'partition_id': str(i),
+                'offset': 2,
+                'sequence_number': 20
+            }
+            checkpoint_store.update_checkpoint(checkpoint)
+
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == partition_cnt
+    for checkpoint in checkpoint_list:
+            assert checkpoint['offset'] == '2'
+            assert checkpoint['sequence_number'] == 20
+
+    checkpoint = {
+                'fully_qualified_namespace': fully_qualified_namespace,
+                'eventhub_name': eventhub_name,
+                'consumer_group': consumer_group,
+                'partition_id': str(0),
+                'offset': '30',
+                'sequence_number': 42
+    }
+    checkpoint_store.update_checkpoint(checkpoint)
+
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == partition_cnt
+    assert checkpoint_list[0]['offset'] == '30'
+
+    for i in range(partition_cnt):
+            checkpoint = {
+                'fully_qualified_namespace': fully_qualified_namespace,
+                'eventhub_name': eventhub_name,
+                'consumer_group': consumer_group,
+                'partition_id': str(i),
+                'offset': 29,
+                'sequence_number': 42
+            }
+            checkpoint_store.update_checkpoint(checkpoint)
+
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == partition_cnt
+    for checkpoint in checkpoint_list:
+            assert checkpoint['offset'] == '29'
+            assert checkpoint['sequence_number'] == 42
```

Comment:
what is the difference b/w 128-152 and this? It seems like only the offset and sequence number were changed? If that's the case, I think you can keep the first set of assertions, but don't need the rest. It's a little redundant b/c it tests what has already been tested.

Reply:
so from line 138-153 it make sure that it accepts when the type of offset is a string rather than a int.  from 155-173 it make sure that a checkpoint is able to update when the partition key and row key exist. what I can do is remove the test case from 155-173 since 138-153 tests that it also able to update when the partition key and row key exists.